### PR TITLE
Add separate metrics server for API

### PR DESCRIPTION
**Pull request by kruai**
# Overview

This PR is being created as part of [ESSNTL-761](https://issues.redhat.com/browse/ESSNTL-761).
It makes it so the metrics endpoint is separate from the other API traffic, and is configurable via Clowder.
I also added a small update to the MQ service's config to make its metrics path configurable as well (only the port was configured before).

## PR Checklist

- [x] Keep PR title short, ideally under 72 characters
- [ ] Descriptive comments provided in complex code blocks
- [ ] Tests: validate optimal/expected output
- [ ] Tests: validate exceptions and failure scenarios
- [ ] Tests: edge cases
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [ ] Links to related PRs

## Secure Coding Practices Documentation Reference

You can find documentation on this checklist [here](https://github.com/RedHatInsights/secure-coding-checklist).

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [x] General Coding Practices

### CODE REVIEW

**Comment by kruai** [comment]
It's unclear to me why the build is failing, but it appears to be caused by connection issues in the ephemeral env :/

**Review by thearifismail — COMMENTED** [review]
LGTM.  How has it been tested?

File: app/__init__.py [hunk 1]
```diff
@@ -203,13 +203,13 @@ def set_request_id():
...
     if runtime_environment.metrics_endpoint_enabled:
-        PrometheusMetrics(
+        metrics = PrometheusMetrics(
             flask_app,
             defaults_prefix="inventory",
             group_by="url_rule",
-            path=None,
             excluded_paths=["^/metrics$", "^/health$", "^/version$", r"^/favicon\.ico$"],
         )
+        metrics.start_http_server(app_config.metrics_port, endpoint=app_config.metrics_path)
```
Comment (thearifismail) [inline app/__init__.py:212]: Wondering how will this work; the `metrics` is created with `flask_app` and then kicks off another using `start_http_server`.  Seeing in action should tell.

**Review by thearifismail — APPROVED** [review]
please see the comment, if anything can be done before merging, else lets merge it and findout.

File: app/__init__.py (hunk 1 above)
Comment (kruai) [reply app/__init__.py:212]: Yeah, it's hard to tell right now. It doesn't complain or break anything right now, but I'm not sure how to test this locally.

**Comment by kruai** [comment]
> LGTM. How has it been tested?

I haven't been able to test that the metrics scraping works correctly; however, I've run the unit tests and some manual tests, and it's at least not breaking anything. If you don't see anything wrong here, I think we should merge and see how it looks in Stage.

**Pull request merged**